### PR TITLE
Adds DeltaSharingOptions

### PR DESCRIPTION
**Pull request by linzhou-db**
Adds DeltaSharingOptions, to support DeltaSharingSource in the next PR.

Moves all options related logic from DeltaSharingDataSource to DeltaSharingOptions.

### CODE REVIEW

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  val CDF_START_TIMESTAMP = "startingTimestamp"
+  val CDF_END_VERSION = "endingVersion"
+  val CDF_END_TIMESTAMP = "endingTimestamp"
+  val CDF_READ_OPTION = "readChangeFeed"
+  val CDF_READ_OPTION_LEGACY = "readChangeData"
+
+  val TIME_TRAVEL_VERSION = "versionAsOf"
+  val TIME_TRAVEL_TIMESTAMP = "timestampAsOf"
+
+  val validOptionKeys : Set[String] = Set(
```
[the remark this answers is not in the document]
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:147]: removed as this is not needed in our case. It was used in DeltaOptions to log the invalid keys passed in, but we don't have such log in the delta sharing client.

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    CDF_READ_OPTION_LEGACY,
+    CDF_START_TIMESTAMP,
+    CDF_END_TIMESTAMP,
+    CDF_START_VERSION,
+    CDF_END_VERSION,
+    "queryName",
+    "checkpointLocation",
+    "path",
+    "timestampAsOf",
+    "versionAsOf"
```
[the remark this answers is not in the document]
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:162]: removed this list as it is not used.

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+// scalastyle:off import.ordering.noEmptyLine
+import java.util.Locale
+
+import scala.util.Try
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.network.util.{ByteUnit, JavaUtils}
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+
+trait DeltaSharingOptionParser {
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:28]: This seems a bit odd. The parser trait does not seem to do any parsing. 
Do you plan to add some parsing utilities here?
Should we call this DeltaSharingOptionsBase instead?

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    if (readChangeFeed) {
+      validCdfOptions.filter(option => options.contains(option._1)).map(option =>
+        option._1 -> options.get(option._1).get
+      )
+    } else {
+     Map.empty[String, String]
+    }
+  }
+
+  private def provideOneStartingOption(): Unit = {
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:94]: Nit: consider calling this: validateOneStartingOption

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    if (startingTimestamp.isDefined && startingVersion.isDefined) {
+      throw DeltaSharingErrors.versionAndTimestampBothSetException(
+        STARTING_VERSION_OPTION,
+        STARTING_TIMESTAMP_OPTION)
+    }
+  }
+
+  private def provideOneTimeTravelOption(): Unit = {
```
Comment (chakankardb) [inline spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:102]: Nit: consider callign this validate*

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala (hunk 4 above)
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:94]: updated.

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala (hunk 5 above)
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:102]: updated.

File: spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala (hunk 3 above)
Comment (linzhou-db) [reply spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala:28]: The `toBoolean` function actually parses the input parameter value, no plan to add other parsing utilities. I feel that's why it's called parser.

**Review by chakankardb — APPROVED** [review]

**Pull request merged**